azerothcore/azerothcore-wotlk
# FIX(CORE/UNIT) Dodges, parries and critters not increasing weapon skills

**Description:**
This fixes not getting skill ups from attacking an enemy and them parrying your attack or dodging it.

This also fixes attacking a critter and not getting weapon skill ups.

Previously Dodge and Parry were only giving a chance to increase the victims DEFENSE skill.
I changed it so it will proc on victim and attacker. for weapon and defense skills

Also critters were being checked to not give weapon skill ups on purpose but I removed the check.



## Changes Proposed:
-  Make it so dodges and parries have a chance to increase weapon skill not just defense skill
-  Make it so critters have a chance to increase weapon skill.
as far as I know there is no way to abuse this because there is not mass critters in one location.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/5882


## How to Test the Changes:


1. Make a nightelf 
2. .character level 20
3. .go 100 100 100
4. .cheat god
5. Find a spider
6. Auto attack it 
7. Observe skill ups when it parries or dodges
8. Find a critter
9. attack it
10. Observe skill up

## Known Issues and TODO List:

Not really I analyzed the entire skill ups algorithm and it follows blizzlike behaviour, and works well enough to not modify at the moment.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
